src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -792,121 +780,24 @@ public File getWebAppRoot() throws Exception {
      * that we need for testing.
      */
     protected ServletContext createWebServer2() throws Exception {
```

Comment:
This method is still `protected`, but subclasses are highly discouraged. The only remaining subclasses are in Jenkins core's test suite as well as a single CloudBees proprietary plugin. For the most part, they represent technical debt and should be replaced with `RealJenkinsRule`.